spec/compiler/normalize/select_spec.cr

```diff
@@ -0,0 +1,68 @@
+require "../../spec_helper"
+
+describe "Normalize: case" do
+  it "normalizes select with call" do
+    assert_expand "select; when foo; body; when bar; baz; end", <<-CODE
+      __temp_1, __temp_2 = ::Channel.select({foo_select_action, bar_select_action})
```

Comment:
Sounds good, but I prefer to wait until we have true parallelism. I don't think any action could be put inside a `select`, there will be mutexes and locks involved and it will be highly coupled with how the runtime works. I'm not sure there will be many extensions to this, we just did it this way so we could more easily implement all the operations.

In any case, this will probably evolve and change in the future until we stabilize everything in 1.0. But I'm convinced that using these things outside a `select` expression makes little sense, so the real names aren't very important, and if they are called `SelectAction` then it's pretty clear it's for them to be used in a `select`.